File: app/src/main/java/com/doubean/ford/ui/groups/groupsHome/GroupsHomeViewModelFactory.java
package com.doubean.ford.ui.groups.groupsHome;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.doubean.ford.data.repository.GroupRepository;
import com.doubean.ford.data.repository.GroupsFollowsAndSavesRepository;

public class GroupsHomeViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private GroupRepository repository;
    private GroupsFollowsAndSavesRepository groupsFollowsAndSavesRepository;

    public GroupsHomeViewModelFactory(@NonNull GroupRepository repository, GroupsFollowsAndSavesRepository groupsFollowsAndSavesRepository) {
        super();
        this.repository = repository;
        this.groupsFollowsAndSavesRepository = groupsFollowsAndSavesRepository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new GroupsHomeViewModel(repository, groupsFollowsAndSavesRepository);
    }
}

File: app/src/main/java/com/doubean/ford/ui/groups/groupsHome/GroupsHomeViewModelFactory.kt
package com.doubean.ford.ui.groups.groupsHome

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.doubean.ford.data.repository.GroupRepository
import com.doubean.ford.data.repository.GroupsFollowsAndSavesRepository

class GroupsHomeViewModelFactory(
    private val repository: GroupRepository,
    private val groupsFollowsAndSavesRepository: GroupsFollowsAndSavesRepository
) : ViewModelProvider.NewInstanceFactory() {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return GroupsHomeViewModel(repository, groupsFollowsAndSavesRepository) as T
    }
}